File: snippets/write-ply.py
def write_ply_lines(lines, out_path):
    strings = ["ply", "format ascii 1.0", "element vertex {}".format(len(lines)),
               "property float x", "property float y", "property float z",
               "property uint8 red", "property uint8 green", "property uint8 blue", "end_header"]
    strings.extend([" ".join([str(x) for x in line]) for line in lines])
    text = "\n".join(strings)
    with open(out_path, "w") as f:
        f.write(text)
        f.write("\n")

File: snippets/write-ply.cpp
void ply_header(stringstream &os, int size)
{
    os << "ply" << endl;
    os << "format ascii 1.0" << endl;
    os << "element vertex " << size << endl;
    os << "property float x" << endl;
    os << "property float y" << endl;
    os << "property float z" << endl;
    os << "property uint8 red" << endl;
    os << "property uint8 green" << endl;
    os << "property uint8 blue" << endl;
    os << "end_header" << endl;
}

void writeStr(std::string path, const std::string &str)
{
    ofstream file;
    file.open(path);
    file << str;
    file.close();
}

void writePly(std::string path, const std::string &lines, int size)
{
    stringstream header;
    ply_header(header, size);
    ofstream plyFile;
    plyFile.open(path);
    plyFile << header.str();
    plyFile << lines;
    plyFile.close();
}

void writePly(std::string path, const std::vector<std::string> &lines)
{
    stringstream ss_buf;
    for (auto &line : lines) {
        ss_buf << line << endl;
    }
    writePly(path, ss_buf.str(), lines.size());
}
